test/test_browser.py

```diff
@@ -5549,6 +5554,69 @@ def test_webpack(self):
     shutil.copyfile('webpack/src/hello.wasm', 'webpack/dist/hello.wasm')
     self.run_browser('webpack/dist/index.html', '/report_result?exit:0')
 
+  def test_fetch_polyfill_shared_lib(self):
+    create_file('library.c', r'''
+      #include <stdio.h>
+      int library_func() {
+        return 42;
+      }
+    ''')
+    create_file('main.c', r'''
+      #include <assert.h>
+      #include <dlfcn.h>
+      #include <stdio.h>
+      #include <emscripten.h>
+      int main() {
+        int found = EM_ASM_INT(
+          return preloadedWasm['/library.so'] !== undefined;
+        );
+        void *lib_handle = dlopen("/library.so", RTLD_NOW);
+        typedef int (*voidfunc)();
+        voidfunc x = (voidfunc)dlsym(lib_handle, "library_func");
+        printf("Got val: %d\n", x());
+        assert(x() == 42);
+        return 0;
```

Comment:
Just return 42 here and then it will show up as `/report_result?42` below I think.  No need for a custom shell html.

Reply:
Interesting! I'll give it a shot.

Reply:
I wasn't able to get a result in the test without using the custom shell. I reduced the HTML to the bare-minimum.

Reply:
If I remove the `'--shell-file', 'report_result.html'` and return from `main()` then the test just hangs:

```
Test suites:
['test_browser']
Running test_browser: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
[Browser harness server on process 3558299]
common:INFO: No EMTEST_BROWSER set. Defaulting to `google-chrome`
common:INFO: Launching browser: ['google-chrome']

Running the browser tests. Make sure the browser allows popups from localhost.

test_fetch_polyfill_shared_lib (test_browser.browser.test_fetch_polyfill_shared_lib) ... Opening in existing browser session.
[unresponsive tests: 1]
common:INFO: Restarting browser process
common:INFO: Launching browser: ['google-chrome']
[test error (see below), automatically retrying]
Expected to find '/report_result?1
' in '[no http server activity]
', diff:

--- expected
+++ actual
@@ -1 +1 @@
-/report_result?1
+[no http server activity]
```

Reply:
Can you try adding `-sEXIT_RUNTIME`?

Reply:
> Can you try adding `-sEXIT_RUNTIME`?

That worked!!!